File: automation/yap/lib/yap-targets.mjs
/**
 * Canonical YAP destinations (where each Giorgio datum belongs).
 * writer: what inserts today | worker: Giorgio worker implementation
 */

export const WRITERS = {
  GIORGIO_WORKER: "giorgio_worker",
  GIORGIO_PLANNED: "giorgio_worker_planned",
  YAP_AUTO: "yap_automatismo",
  OPERATORE: "operatore_yap",
};

export const YAP_NAV = {
  agendaPopup: "Agenda > double click appointment > popup 'Dettagli appuntamento'",
  agendaBar: "Agenda > event bar (title composed by YAP)",
  praticaFromPopup: "Appointment popup > link 'Gestione pratica'",
  praticaOverview: "Gestione pratica > overview (PraticaGetOverviewAction)",
  praticaRevisione: "Gestione pratica > menu 'Revisione'",
  praticaPreventivi: "Gestione pratica > menu 'Preventivi'",
  praticaOdl: "Gestione pratica > menu 'Ordini di lavoro'",
  praticaMateriali: "Gestione pratica > menu 'Materiali di consumo'",
  praticaSmaltimento: "Gestione pratica > menu 'Smaltimento rifiuti'",
  praticaDocumenti: "Gestione pratica > menu 'Documenti fiscali'",
  odlManodopera: "Ordini di lavoro > MANODOPERA rows (PropertyGetAction catalog)",
  odlMateriali: "Ordini di lavoro > MATERIALI DI CONSUMO",
  odlArticoli: "Ordini di lavoro > warehouse items / document rows",
  odlDeleteToolbar: "Ordini di lavoro > toolbar 'Elimina'",
  odlDeleteConfirm: "Ordini di lavoro > confirm 'Confermi di voler eliminare l'ordine di lavoro?'",
};

/** Agenda popup fields (observed UI order). */
export const AGENDA_POPUP_FIELDS = {
  cosa: { id: "cosa", label: "Cosa", index: 1, writer: WRITERS.GIORGIO_WORKER, worker: "implemented" },
  quando: { id: "quando", label: "Quando", writer: WRITERS.GIORGIO_WORKER, worker: "implemented" },
  dalle: { id: "dalle", label: "dalle", writer: WRITERS.GIORGIO_WORKER, worker: "implemented" },
  alle: { id: "alle", label: "alle", writer: WRITERS.GIORGIO_WORKER, worker: "implemented" },
  note1: { id: "note1", label: "note (field 1)", writer: WRITERS.GIORGIO_PLANNED, worker: "planned", env: "YAP_FILL_NOTES=1" },
  note2: { id: "note2", label: "note (field 2)", writer: WRITERS.GIORGIO_PLANNED, worker: "planned" },
  tag: { id: "tag", label: "Tag (chip)", writer: WRITERS.GIORGIO_WORKER, worker: "implemented" },
};

export const AUTOMATISMI = {
  odlDaPrenotazione: "AutomatismoOdlDaPrenotazione",
  revisioneInPrenotazione: "AutomatismoRevisioneInPrenotazione",
  articoliDaTag: "AutomatismoArticoloDocumentoFromTagPrenotazione",
};

export function writerLabel(writer) {
  const map = {
    [WRITERS.GIORGIO_WORKER]: "Giorgio worker",
    [WRITERS.GIORGIO_PLANNED]: "Giorgio worker (planned)",
    [WRITERS.YAP_AUTO]: "YAP automation",
    [WRITERS.OPERATORE]: "YAP operator",
  };
  return map[writer] || writer;
}

File: backend/yap_targets.py
"""Destinazioni YAP canoniche (allineate a automation/yap/lib/yap-targets.mjs)."""

from __future__ import annotations

WRITERS = {
    "GIORGIO_WORKER": "giorgio_worker",
    "GIORGIO_PLANNED": "giorgio_worker_planned",
    "YAP_AUTO": "yap_automatismo",
    "OPERATORE": "operatore_yap",
}

YAP_NAV = {
    "agenda_popup": "Agenda › doppio click appuntamento › popup «Dettagli appuntamento»",
    "agenda_bar": "Agenda › barra evento (titolo composto da YAP)",
    "pratica_from_popup": "Popup appuntamento › link «Gestione pratica»",
    "pratica_overview": "Gestione pratica › overview (PraticaGetOverviewAction)",
    "pratica_revisione": "Gestione pratica › menu «Revisione»",
    "pratica_preventivi": "Gestione pratica › menu «Preventivi»",
    "pratica_odl": "Gestione pratica › menu «Ordini di lavoro»",
    "pratica_materiali": "Gestione pratica › menu «Materiali di consumo»",
    "pratica_smaltimento": "Gestione pratica › menu «Smaltimento rifiuti»",
    "odl_manodopera": "Ordini di lavoro › righe MANODOPERA",
    "odl_materiali": "Ordini di lavoro › MATERIALI DI CONSUMO",
    "odl_articoli": "Ordini di lavoro › articoli magazzino",
}

AUTOMATISMI = {
    "odl": "AutomatismoOdlDaPrenotazione",
    "revisione": "AutomatismoRevisioneInPrenotazione",
    "articoli": "AutomatismoArticoloDocumentoFromTagPrenotazione",
}

WRITER_LABELS = {
    WRITERS["GIORGIO_WORKER"]: "Giorgio worker",
    WRITERS["GIORGIO_PLANNED"]: "Giorgio worker (pianificato)",
    WRITERS["YAP_AUTO"]: "Automatismo YAP",
    WRITERS["OPERATORE"]: "Operatore in YAP",
}


def writer_label(writer: str) -> str:
    return WRITER_LABELS.get(writer, writer)
